react/states_and_effects/how_to_deal_with_side_effects.md

```diff
@@ -0,0 +1,242 @@
+### Introduction
+
+Certain components in react need to be interacting with other things outside the component itself. These other things can be anything from querying data from a server to synchronizing with the position of the component on the webpage or even sending some data to a server when necessary. This leads to the arising of a new concept called 'effects'. 
+
+While we are already familiar with rendering code and adding event handlers, it is not enough for all uses, like when you want to connect to your server and fetch messages to show to a user. Effects let you run some code to synchronize your component as necessary, on rendering or a reactive/state value change rather than on a particular event.
+
+Similar to how we have the `useState` hook, React offers us a handy `useEffect` hook to use effects in our components.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   What are effects?
+*   How are effects used in React?
+*   What are the different parts of an `useEffect` hook?
+*   When should I use an effect?
+
+### Using Effect Saves The Day
+
+#### The useEffect Hook
+
+Let us take a component in question. We want to make a "Clock" component that shows how many seconds have passed since the user has loaded the webpage. To update it every second, we can use our nifty `setInterval` function to add one to the `counter` state variable, every second. Let's try putting it in the body of our component.
+
+~~~jsx
+import React, { useState } from "react";
+
+export default function Clock() {
+  const [counter, setCounter] = useState(0);
+
+  setInterval(() => {
+    setCounter(count => count + 1)
+  }, 1000);
+
+  return (
+    <p>{counter} seconds have passed.</p>
+  );
+}
+~~~
+
+Alas, we see our counter going berserk. The reason this occurs is that we try to **manipulate the state during render**. As we know, the component gets torn down and re-rendered everytime the state updates, and we are updating the state every second, thus incrementing rapidly.
+
+This is where the `useEffect` hook swoops in to save us. We can wrap this calculation inside an `useEffect` hook to move it outside the rendering calculation. It accepts a callback function with all the calculations.
+
+~~~jsx
+import React, { useEffect, useState } from "react";
+
+export default function Clock() {
+  const [counter, setCounter] = useState(0);
+
+  useEffect(() => {
+    setInterval(() => {
+      setCounter(count => count + 1)
+    }, 1000);
+  })
+
+  return (
+    <p>{counter} seconds have passed.</p>
+  );
+}
+~~~
+
+#### The Dependency Array
+
+But, it still keeps growing too fast! This is where another argument of the `useEffect` comes in: the dependency array.
```

Comment:
I think moving this paragraph makes the transition to the next section more smooth
```suggestion
But, it still keeps growing too fast! This is where another argument of the `useEffect` comes in: the dependency array.

#### The Dependency Array
```